python/res/simulator/batch_simulator.py

```diff
@@ -204,9 +204,6 @@ def start(self, case_name, case_data):
         sim_context = BatchContext(
             self.result_keys, self.ert, file_system, mask, itr)
 
-        for sim_id, (geo_id, _) in enumerate(case_data):
```

Comment:
This scares me a bit. Could you please try to build and run Everest tests on top of this?

Reply:
In particular, the `geo_id` is lost.

Reply:
now `case_data` is an extra argument to `simulation_context`